contracts/internal/EverlongStorage.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.22;
+
+import { DoubleEndedQueue } from "openzeppelin/utils/structs/DoubleEndedQueue.sol";
+
+// TODO: Reassess whether centralized configuration management makes sense.
+//       https://github.com/delvtech/everlong/pull/2#discussion_r1703799747
+/// @author DELV
+/// @title EverlongStorage
+/// @notice Base contract for Everlong.
+/// @custom:disclaimer The language used in this code is for coding convenience
+///                    only, and is not intended to, and does not, have any
+///                    particular legal or regulatory significance.
+abstract contract EverlongStorage {
```

Comment:
As a general comment, I'm confused about why some things are public and others aren't. I understand why things like `asset` and `decimals` are internal since we need to override these getters due to the use of `ERC4626`, but there are several other fields that live entirely within Everlong that could be made public.

Reply:
Good point, not super happy with how things are currently structured and was able to isolate this a bit better as part of the vault share pricing work. Will be cleaned up in next pr :)